CHANGELOG.md

```diff
@@ -8,6 +8,17 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased] - ReleaseDate
 
+### Breaking Changes
+
+- query: `RowBinaryWithNamesAndTypes` is now used by default for query results. This may cause panics if the row struct definition does not match the database schema. Use `Client::with_validation(false)` to revert to the previous behavior which uses plain `RowBinary` format for fetching rows. ([#221])
+- query: due to `RowBinaryWithNamesAndTypes` format usage, there might be an impact on fetch performance, which largely depends on how the dataset is defined. If you experience performance issues, consider disabling validation by using `Client::with_validation(false)`.
+
+### Added
+- client: added `Client::with_validation` builder method. Validation is enabled by default, meaning that `RowBinaryWithNamesAndTypes` format will be used to fetch rows from the database. If validation is disabled, `RowBinary` format will be used, similarly to the previous versions. ([#221]).
+- types: a new crate `clickhouse-types` was added to the project workspace. This crate is required for `RowBinaryWithNamesAndTypes` struct definition validation, as it contains ClickHouse data types AST, as well as functions and utilities to parse the types out of the ClickHouse server response. Note that this crate is not intended for public usage, as it might introduce internal breaking changes not following semver. ([#221]).
```

Comment:
I think we can follow semver in `clickhouse-types`, so let's remove this additional note.